File: frontend/types/projects.ts
import type { JobStatus, VideoStyle } from "@/types";

// GET /api/projects
export interface ProjectItem {
  job_id: string;
  status: JobStatus;
  title: string;
  topic: string;
  style: VideoStyle;
  duration: number | null;
  elapsed_seconds: number | null;
  created_at: string;
  error: string | null;
  thumbnail_url: string | null;
  video_url: string | null;
  download_url: string | null;
}

// GET /api/health
export interface HealthInfo {
  status: string;
  ffmpeg: { found: boolean; version: string | null };
  config: {
    groq_api_key_set: boolean;
    pexels_api_key_set: boolean;
    groq_model: string;
    tts_provider: string;
    render_size: string;
  };
  warnings: string[];
}

// POST /api/ideas
export interface IdeasResponse {
  ideas: string[];
}

File: backend/app/models/projects.py
from pydantic import BaseModel

from app.models.jobs import JobStatus
from app.models.schemas import VideoStyle


class ProjectItem(BaseModel):
    job_id: str
    status: JobStatus
    title: str
    topic: str
    style: VideoStyle
    duration: float | None = None
    elapsed_seconds: float | None = None
    created_at: str
    error: str | None = None
    thumbnail_url: str | None = None  # sirf completed videos ke liye
    video_url: str | None = None
    download_url: str | None = None
